Initial implementation draft

jcechace opened the pull request.

jcechace committed:
Initial implementation draft

pestanko commented on bin/console:
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env ruby
```
File can be deleted

jcechace replied:
Sure.. on the other hand I use it for testing purposes -- it starts ruby repl and loads the module right away.

pestanko commented on bin/setup:
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
```
File can be deleted

pestanko replied:
Neither this one.

pestanko commented on lib/rb_shift/client.rb:
```diff
@@ -0,0 +1,78 @@
+      @token      = bearer_token
+      @url        = url
+      @kubernetes = RestClient::Resource.new "#{url}/api/v1",
+                                             verify_ssl: OpenSSL::SSL::VERIFY_NONE,
```
This should be parametrized

jcechace replied:
Agreed, SSL validation should be a subject to configuration.

pestanko commented on bin/console:
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env ruby
```
This file is not required I think.

mijaros committed:
Rbshift now properly uses oc cli

Adding replication controller to check whether deployment
is running of completed.

mijaros committed:
Adds support for inplace reload of object

* The method reload is added to OpenshiftKind
* This is usable for inplace reload of object
  without need to access or invalidate parent element

Signed-off-by: Miroslav Jaros <mirek@mijaros.cz>

mijaros committed:
Fix missing comment in Client

Fix misused unless in service::create_route

Signed-off-by: Miroslav Jaros <mirek@mijaros.cz>

jcechace committed:
Updated rubocop config

jcechace committed:
Updated project files

jcechace committed:
Resolving some rubocop violations

mijaros committed:
Remove unused unfold_params method

Replace detect with find

mijaros committed:
Change unfold_opts method to use case instead of conditionals

Modify client initialization to use either bearer_token or username,
password tuple
Fix some rubocop issues

Signed-off-by: Miroslav Jaros <mirek@mijaros.cz>

mijaros committed:
Add custom ```InvalidAuthorizationError```

Fix token attribute setting in client

Signed-off-by: Miroslav Jaros <mirek@mijaros.cz>

jcechace commented:
waiting for #2
